CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerHealthComponent.cs

```diff
@@ -35,6 +35,19 @@ public void TakeDamage(int damage)
         }
     }
 
+    public void TakeBrainDamage(int damage)
+    {
+        if (isDead) return;
+        DecreaseHealthBy(damage);
+        _effect.ShakeScreen();
+        _effect.StartCoroutine(nameof(_effect.FlashFX));
+        if (currentHealth <= 0)
+        {
+            currentHealth = 0;
+            Die();
+        }
+    }
+
```

Comment:
this is nearly an identical function to the one above except that it has one single OR for invincibility removed. Make a core function for the core functionality and call it in both.